lib/active_merchant/billing/gateways/cashnet.rb

```diff
@@ -117,11 +117,11 @@ def add_invoice(post, money, options)
           codes_and_amounts = options[:item_codes].transform_keys { |key| key.to_s.delete('_') }
           codes_and_amounts.each do |key, value|
             post[key] = value if key.start_with?('itemcode')
-            post[key] = amount(value) if key.start_with?('amount')
+            post[key] = amount(value.to_i) if key.start_with?('amount')
```

Comment:
The amount method seems to be then transforming this back into a string 🤦🏻  This was discovered by adding a puts statement to confirm the class. I think instead that the fix would be `amount(money).to_i`